x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessor.java

```diff
@@ -288,9 +288,15 @@ void processResult(AutodetectResult result) {
             // Commit previous writes here, effectively continuing
             // the flush from the C++ autodetect process right
             // through to the data store
-            bulkResultsPersister.executeRequest();
-            persister.commitResultWrites(jobId);
-            flushListener.acknowledgeFlush(flushAcknowledgement);
+            try {
+                bulkResultsPersister.executeRequest();
+                persister.commitResultWrites(jobId);
+            } catch (Exception e) {
+                LOGGER.error("[" + jobId + "] failed to bulk persist results and commit writes during flush acknowledgement.", e);
```

Comment:
I think it should rethrow the exception here, otherwise this PR is doing more than simply fixing the flush problems - it would be straying into the territory of handling result persistence problems, which I think we should address in a separate PR.  We're not eating persistence exceptions in other places in this method.

Reply:
👍